src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -636,42 +658,73 @@ private bool StartCore(ProcessStartInfo startInfo)
 
                     if (startInfo.UserName.Length != 0)
                     {
+                        if (startInfo.Password != null && startInfo.PasswordInClearText != null)
+                        {
+                            throw new ArgumentException(SR.CantSetDuplicatePassword);
+                        }
+
                         Interop.Advapi32.LogonFlags logonFlags = (Interop.Advapi32.LogonFlags)0;
                         if (startInfo.LoadUserProfile)
                         {
                             logonFlags = Interop.Advapi32.LogonFlags.LOGON_WITH_PROFILE;
                         }
 
-
-                        try { }
-                        finally
+                        IntPtr passwordPtr = IntPtr.Zero;
+                        try
                         {
-                            retVal = Interop.Advapi32.CreateProcessWithLogonW(
-                                    startInfo.UserName,
-                                    startInfo.Domain,
-                                    startInfo.PasswordInClearText,
-                                    logonFlags,
-                                    null,            // we don't need this since all the info is in commandLine
-                                    commandLine,
-                                    creationFlags,
-                                    environmentPtr,
-                                    workingDirectory,
-                                    startupInfo,        // pointer to STARTUPINFO
-                                    processInfo         // pointer to PROCESS_INFORMATION
-                                );
+                            if (startInfo.Password != null)
+                            {
+                                passwordPtr = Marshal.SecureStringToGlobalAllocUnicode(startInfo.Password);
+                            }
+                            else
+                            {
+                                unsafe
+                                {
+                                    fixed (char* p = startInfo.PasswordInClearText ?? string.Empty)
+                                    {
+                                        passwordPtr = (IntPtr)p;
+                                    }
```

Comment:
This is not correct.  The string needs to be fixed for the duration of its usage.